src/offset/mod.rs

```diff
@@ -206,6 +206,27 @@ pub trait TimeZone: Sized + Clone {
     /// The original `TimeZone` value can be recovered via `TimeZone::from_offset`.
     type Offset: Offset;
 
+    /// Make a new `DateTime` from year, month, day, time components and current time zone.
+    ///
+    /// This assumes the proleptic Gregorian calendar, with the year 0 being 1 BCE.
+    ///
+    /// Returns `LocalResult::None` on invalid input data.
+    fn with_ymd_and_hms(
```

Comment:
This looks good, but its a reminder for me that we need to look into moving to a regular `Result<DateTime<Self>, Error` where the local result could be one variant in the error

Reply:
I agree, but IMO that should only happen on the main branch where we can also change all the `_opt()` methods to lose the suffix and then return a `Result`.

Reply:
yep it could also partially be covered by: https://github.com/chronotope/chrono/pull/830